wafer/schedule/admin.py

```diff
@@ -218,23 +245,12 @@ class ScheduleItemAdmin(admin.ModelAdmin):
     def changelist_view(self, request, extra_context=None):
         extra_context = extra_context or {}
         # Find issues in the schedule
-        all_items = None
-        clashes = find_clashes()
-        validation = validate_items(all_items)
-        venues = find_invalid_venues()
-        duplicates = find_duplicate_schedule_items(all_items)
-        non_contiguous = find_non_contiguous(all_items)
+        all_items = prefetch_schedule_items()
         errors = {}
-        if clashes:
-            errors['clashes'] = clashes
-        if duplicates:
-            errors['duplicates'] = duplicates
-        if validation:
-            errors['validation'] = validation
-        if venues:
-            errors['venues'] = venues
-        if non_contiguous:
-            errors['non_contiguous'] = non_contiguous
+        for validator, err_type, _msg in SCHEDULE_ITEM_VALIDATORS:
+            failed_items = validator(all_items)
+            if failed_items:
+                errors[err_type] = failed_items
```

Comment:
I'm a little worried this can override errors if `err_type` is not unique (and the registration interface doesn't prevent that). Maybe we could do:
```python
errors = defaultdict(list)
...
errors[err_type].extend(failed_items)
```
?